pkg/reconciler/knativeserving/controller.go

```diff
@@ -40,7 +40,7 @@ const (
 var (
 	recursive  = flag.Bool("recursive", false, "If filename is a directory, process all manifests recursively")
 	MasterURL  = flag.String("master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
-	Kubeconfig = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+	Kubeconfig = flag.String("kubeconfig", os.Getenv("KUBECONFIG"), "Path to a kubeconfig. Only required if out-of-cluster.")
```

Comment:
Can we actually remove these altogether? They should come via sharedmain anyway, I think? (modulo recursive)

Reply:
How can we access these arguments from sharedmain?
```
config, err := mf.NewManifest(filepath.Join(koDataDir, "knative-serving/"), *recursive, cfg)
```
To create Manifest, we need the cfg. The cfg is created by 
```
cfg, err := clientcmd.BuildConfigFromFlags(*MasterURL, *Kubeconfig)
```
so we need MasterURL and Kubeconfig. How can we access them in sharedmain?

Reply:
You're right indeed, we can't reuse them cause they are captured in `MainWithConfig`.

I'd propose to make these flags package local (lowercase), use the plain `Main` function in our `cmd/manager/main.go` and reuse `sharedmain.GetConfig(*masterURL, *kubeconfig)` here locally. That function does all the defaulting needed by all controllers.

Reply:
Is it sharedmain.GetConfig("", "")?

Reply:
well, if you want to keep supporting the flags you gotta pass them in.

Reply:
@markusthoemmes I made some changes.